test/chunking-form/samples/mixed-synthetic-named-exports/_expected/es/main.js

```diff
@@ -0,0 +1,16 @@
+const d = {
+  fn: 42,
+  hello: 'hola'
+};
+const foo = 100;
+
+var ns = /*#__PURE__*/Object.freeze(Object.assign({
+  __proto__: null,
+  foo: foo
+}, d));
```

Comment:
Nice with the Object.assign 👍
However, I think the default export is missing in the namespace?

Reply:
If don't skip default, basically means we always have to generate the Object.assing() since there is always a default.
- If synthetic named exports exists, the "default" export becomes the namespace.

```
export default {
  'default': 12
}
```

```
import {default} from './mod';
default === 12
```
- It would be great to generate:
```
const ns = default;
```
without additional runtime if there are not other named exports.

For commonjs it would also help to skip the "default" named export, this there is not such a thing in the original cjs module.

Reply:
I left my comment below.  In short yes, I see your point, but unless we can guarantee that the default export already contains itself as property "default" (which would be rather uncommon for CJS code), this will break ES module semantics.

Reply:
By the way, at the moment we go to even greater lengths to ensure proper semantics when dynamically importing an external file for CJS output:

https://rollupjs.org/repl/?version=1.27.13&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCgnZXh0ZXJuYWwnKS50aGVuKG5zJTIwJTNEJTNFJTIwY29uc29sZS5sb2cobnMuZGVmYXVsdCkpJTNCJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

This helper takes whatever is returned by `require(external)` and
- attaches it to ns.default
- copies all its properties over as named exports, creating getters to ensure live-bindings.

I do not think we necessarily need to go that far for reified internal namespaces as nobody yet complained about missing live-bindings (maybe at some point we should reconsider...?), but I think we should have the `default` as part of the namespace.

Reply:
A note:

```js
import {default} from './mod';
default === 12
```

`default` is not a legal identifier. What you could write, is

```js
import {default as foo} from './mod';
foo === 12
```

As a matter of fact, Rollup will treat this the same as

```js
import foo from './mod';
foo === 12
```

i.e. there can never be a named export `default`.

https://rollupjs.org/repl/?version=1.27.13&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QmRlZmF1bHQlMjBhcyUyMGZvbyU3RCUyMGZyb20lMjAnZXh0ZXJuYWwnJTNCJTVDbmltcG9ydCUyMGJhciUyMGZyb20lMjAnZXh0ZXJuYWwnJTNCJTVDbiU1Q25jb25zb2xlLmxvZyhmb28lMkMlMjBiYXIpJTNCJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzbSUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE